diff --git a/src/sd-jwt/types.ts b/src/sd-jwt/types.ts
index 3857aa6ea94feba4f6142522010ffd2511a6f3c6..02b793a3202aaeeb6ea8ad47c85f87c4a94e2257 100644
--- a/src/sd-jwt/types.ts
+++ b/src/sd-jwt/types.ts
@@ -49,7 +49,7 @@ export const SdJwt4VC = z.object({
     typ: CredentialFormat,
     alg: z.string(),
     kid: z.string().optional(),
-    x5c: z.string().optional(),
+    x5c: z.array(z.string()).optional(),
     vctm: z.array(z.string()).optional(),
   }),
   payload: z.intersection(
