diff --git a/src/components/buttons/IconButton.tsx b/src/components/buttons/IconButton.tsx
index adf70825b1c8f206a15641a4a1368c247f9b0ea2..eac2b5bc3627a5bc46a0b43512b4f8e2d9359cde 100644
--- a/src/components/buttons/IconButton.tsx
+++ b/src/components/buttons/IconButton.tsx
@@ -122,6 +122,7 @@ export const IconButton = ({
       testID={testID}
     >
       <Animated.View
+        pointerEvents={"none"}
         style={[
           IOIconButtonStyles.button,
           IOIconButtonStyles.buttonSizeSmall,
diff --git a/src/components/buttons/IconButtonSolid.tsx b/src/components/buttons/IconButtonSolid.tsx
index 209a2a37cbe89f3475c6df1f471ee8a68d3a9edf..fb323d095de0e012fac1e487ee2d1debd2f33e8d 100644
--- a/src/components/buttons/IconButtonSolid.tsx
+++ b/src/components/buttons/IconButtonSolid.tsx
@@ -102,6 +102,7 @@ export const IconButtonSolid = ({
       style={{ alignSelf: "flex-start" }}
     >
       <Animated.View
+        pointerEvents={"none"}
         style={[
           IOIconButtonStyles.button,
           IOIconButtonStyles.buttonSizeLarge,
