diff --git a/dist/index.js b/dist/index.js
index 31be845957aa78c319d2cf0646459fe52e7765b7..37c6be77ae9ed9f306fd4015f35a2e2de8066c9c 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -493,6 +493,7 @@ async function createAuthorizationResponse(options) {
       apu: (0, import_io_wallet_utils4.encodeToBase64Url)(nonceBytes),
       apv: (0, import_io_wallet_utils4.encodeToBase64Url)(requestObject.nonce),
       enc,
+      kid: encryptionJwk.kid,
       method: "jwk",
       publicJwk: encryptionJwk
     };
diff --git a/dist/index.mjs b/dist/index.mjs
index 1bd31ab08ed6a4014c308137fcef2b5cbeadf50b..b8bdf4e7da359e1219430a88974df70a4ad2ef9e 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -453,6 +453,7 @@ async function createAuthorizationResponse(options) {
       apu: encodeToBase64Url(nonceBytes),
       apv: encodeToBase64Url(requestObject.nonce),
       enc,
+      kid: encryptionJwk.kid,
       method: "jwk",
       publicJwk: encryptionJwk
     };
